### ISSUE TITLE
Some problems with reimplementing the training process

### ISSUE THREAD

**Issue by Alienge** [Vchitect/Latte#104]
I used the [FaceForensics dataset](https://huggingface.co/datasets/maxin-cn/FaceForensics) to train the Latte XL-2 model. However, I found that the FVD is lower than what is reported in the paper. I want to understand the reason for this discrepancy. I carefully compared the differences in experimental settings and found that the biggest difference is the number of GPU. Due to limited computing resources, I want to know how the number of GPUs influence the results. Below are the details of my experimental settings and FVD comparison results.

### Number of  GPU
|  Paper Reported | Ours|
|  ----  | ----  |
| 8 A100 80G  | 2 A100 80G|

###  FVD :

|  Paper Reported | Ours|
|  ----  | ----  |
| 34.00  | 71.01|

Here is the training log file.  I use the 220000.pt to calculate FVD. 
[log.txt](https://github.com/user-attachments/files/16456477/log.txt)

**Comment by maxin-cn** [comment]
Thanks for your interest. Could you tell me some details about your FVD evaluation? Thanks~

**Comment by Alienge** [comment]
Here ([#issue 80](https://github.com/Vchitect/Latte/issues/80) ). And I also find gradient accumulation is not influence the FVD.

**Comment by maxin-cn** [comment]
I see you've been training for about four days. Did you use ema test or model test (https://github.com/Vchitect/Latte/blob/9ec0fb0fbf0e2f2df754a39e77b783dc68c52443/train.py#L258)? Given your resources, I think you could try model testing.

**Comment by maxin-cn** [comment]
I know of others who have tried to retrain Latte on UCF101 to replicate the results. They trained with an 8 80G A100 GPUs for about a week and were able to get FVD results similar to those reported in the paper. I think you can find it in issues.

**Comment by maxin-cn** [comment]
BTW, I see that your training speed is relatively fast. I would like to know what is your single GPU batchsize?

**Comment by Alienge** [comment]
I use the ema model for test. And the batch size is default 5.  Other config used in default setting.

**Comment by Alienge** [comment]
I guess the number of GPUs will influence the result. But i cannot have enough GPUs to verify

**Comment by maxin-cn** [comment]
You can try it out with model. Also, you might be able to increase the batchsize.

**Comment by maxin-cn** [comment]
I need to check tomorrow if I'm using 5.

**Comment by Alienge** [comment]
Ok, i will test with model. It need about 5 hours. I will report it in the tomorrow too.

**Comment by Alienge** [comment]
Sorry for late reply. I use the "model" weight to inference. The FVD increases to 99.78. Any advices?

**Comment by maxin-cn** [comment]
Did you use the FFS train split to evaluate FVD? If so, continue training and then re-evaluate the FVD.

**Comment by maxin-cn** [comment]
The batch size I used was also 5 for a single GPU.

> I need to check tomorrow if I'm using 5.

**Comment by maxin-cn** [comment]
I think you might use a gradient checkpoint to increase the batch size of a single GPU. Here is the example (https://github.com/Vchitect/Latte/blob/9ec0fb0fbf0e2f2df754a39e77b783dc68c52443/models/latte_t2v.py#L807).

**Comment by Alienge** [comment]
Ok, let me check it and add it to the origin file [ Latte.py](https://github.com/Vchitect/Latte/blob/main/models/latte.py). And finally I confirm I should re-train the model?

**Comment by Alienge** [comment]
> Did you use the FFS train split to evaluate FVD? If so, continue training and then re-evaluate the FVD.
Yes, I use the dir is "/Data/preprocess_ffs/train/videos/"  in ffs_train.yaml. And evaluate FVD using the data from [here](https://disk.yandex.ru/d/v7MS7zu4mmZxXw/datasets)

**Comment by Alienge** [comment]
> Did you use the FFS train split to evaluate FVD? If so, continue training and then re-evaluate the FVD.

Yes, I use the dir is "/Data/preprocess_ffs/train/videos/"  in ffs_train.yaml. And evaluate FVD using the data from [here](https://disk.yandex.ru/d/v7MS7zu4mmZxXw/datasets)

**Comment by maxin-cn** [comment]
> Ok, let me check it and add it to the origin file [ Latte.py](https://github.com/Vchitect/Latte/blob/main/models/latte.py). And finally I confirm I should re-train the model?

You can increase the batch size, resume the model, and continue to train.

**Comment by Alienge** [comment]
> > Ok, let me check it and add it to the origin file [ Latte.py](https://github.com/Vchitect/Latte/blob/main/models/latte.py). And finally I confirm I should re-train the model?
> 
> You can increase the batch size, resume the model, and continue to train.

Ok. I will be reply with result in 3-4 days in here.

**Comment by Alienge** [comment]
> > > Ok, let me check it and add it to the origin file [ Latte.py](https://github.com/Vchitect/Latte/blob/main/models/latte.py). And finally I confirm I should re-train the model?
> > 
> > 
> > You can increase the batch size, resume the model, and continue to train.
> 
> Ok. I will be reply with result in 3-4 days in here.


Hi, Xin. Sorry for late reply. It required huge times to re-train. The gap between your reported in the paper decrease.  Below are the details of FVD comparison results and my experimental log.

###  FVD :

|  Paper Reported | Reimpletemtation|
|  ----  | ----  |
| 34.00  | 48.629|

We also get FVD with different epoch. 
|  Epoch | ema| model| 
|  ----  | ----  |  ----|
| 30k  | 48.629| 64.705 |
| 28k  | 52.836| - |

It seems that continuing to train can further reduce FVD. 

[Here ](https://github.com/user-attachments/files/16557762/log.txt) is the training log file.

Other config used in default setting except local_batch_size = 20.  If I want to get closer to the FVD you mentioned in the paper, do I need to continue training and increase the learning rate?

**Comment by maxin-cn** [comment]
It seems that the FVD value is still decreasing, you can continue to train.
